src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,35 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+
+    PUSH_ARGUMENT_REGISTERS
+
```

Comment:
The stack needs to be 16-byte aligned on Unix - I do not think you are maintaining the alignment here.